src/GitHub.Api/Application/ApiClient.cs

```diff
@@ -198,22 +210,55 @@ public ApiClient(UriString hostUrl, IKeychain keychain, IGitHubClient githubClie
                 await ValidateKeychain();
                 await ValidateCurrentUserInternal();
 
-                GitHubRepository repository;
+                var uriString = keychain.Connections.First().Host;
+                var keychainAdapter = await keychain.Load(uriString);
+
+                var command = new StringBuilder("publish -r ");
+                command.Append(newRepository.Name);
+
+                if (!string.IsNullOrEmpty(newRepository.Description))
+                {
+                    command.Append(" -d ");
+                    command.Append(newRepository.Description);
+                }
+
                 if (!string.IsNullOrEmpty(organization))
                 {
-                    logger.Trace("Creating repository for organization");
+                    command.Append(" -o ");
+                    command.Append(organization);
+                }
 
-                    repository = (await githubClient.Repository.Create(organization, newRepository)).ToGitHubRepository();
+                if (newRepository.Private ?? false)
+                {
+                    command.Append(" -p");
                 }
-                else
+
+                var octorunTask = new OctorunTask(taskManager.Token, nodeJsExecutablePath, octorunScriptPath, command.ToString(), 
+                    user: keychainAdapter.Credential.Username, userToken: keychainAdapter.Credential.Token)
+                    .Configure(processManager);
+
+                var ret = await octorunTask.StartAwait();
+
+                if (ret.Count == 0)
                 {
-                    logger.Trace("Creating repository for user");
+                    throw new ApiClientException("Publish failed");
+                }
 
-                    repository = (await githubClient.Repository.Create(newRepository)).ToGitHubRepository();
+                if (ret[0] == "success")
```

Comment:
Since everyone that's running `OctorunTask` is going to be doing exactly the same `if (ret[0] == "success")` to check for status codes and output, it's probably a good idea if we have an actual object with the data returned from OctorunTask instead of a list of strings perhaps? Manually checking strings is very error prone. Maybe just having a static helper method that everyone calls to convert the output of OctorunTask to something that's properly typed would be the the easiest way, no need to change OctorunTask at all.

Reply:
Addressed in #616